### ISSUE TITLE
[feature] improve documentation for JavaScript API

### ISSUE THREAD

**Issue by ExLineP** [ast-grep/ast-grep#1416]
# ⭐ Suggestion

Currently there is no straightforward way to actually use this api as codemode tool. For example, findInFiles only mentioned is in the Tips, but not in api references, which is weird. And all of the functions are not covered in api docs making it hard to actually use this tool.

**Comment by HerringtonDarkholme** [comment]
Do you mean findInFiles and parseFiles, or the JS APi in general?

> no straightforward way to actually use this api as codemode tool

I'm not too sure what does this mean. https://ast-grep.github.io/guide/api-usage/js-api.html#fix-code are you talking about this page? CodeFix is quite new in the JS API and also it is not AST-sensitive. I wonder how many people would like to use this instead of Babel or other tools.

**Comment by ExLineP** [comment]
I mean, documentation and examples are not properly structured and it's really hard to follow it.

**Comment by HerringtonDarkholme** [comment]
Do you have a more specific suggestion for the structure and examples in mind? 

A bullet list and some code snippet will be helpful
